node_modules/oae-email/lib/test/util.js

```diff
@@ -106,13 +106,17 @@ var collectAndFetchAllEmails = module.exports.collectAndFetchAllEmails = functio
                     ActivityEmail.collectAllBuckets(function() {
                         EmailAPI.removeListener('debugSent', _handleDebugSent);
 
-                        // Ensure no email has an html content line longer than 997 chars (999 chars
-                        // minus CR+LF) to ensure we don't have any issues with postfix line-wrap
+                        // SMTP specifications have a requirement that no line in an email body
+                        // should surpass 999 characters, including the CR+LF character. This check
+                        // ensures no email has an html content line longer than 500 chars, which
+                        // accounts for post-processing things such as SendGrid changing the links
+                        // to extremely long values (e.g., ~400 characters long) for click-tracking
+                        //
                         // @see https://github.com/oaeproject/Hilary/issues/1168
                         _.each(messages, function(message) {
                             assert.ok(_.isString(message.html));
                             _.each(message.html.split('\n'), function(line) {
-                                assert.ok(line.length <= 997, util.format('Expected no email line to be more than 997 characters, but found: %s', line));
+                                assert.ok(line.length <= 500, util.format('Expected no email line to be more than 500 characters, but found: (%s) %s', line.length, line));
```

Comment:
Should we check that a line can't have more than 2 links as sendgrid might push it over 1k?